### PR TITLE
#20 Add custom login form

### LINKED ISSUE #20: Add custom login form
1. Change default login form to this bootstrap template
https://getbootstrap.com/docs/4.0/examples/sign-in/

### PR DESCRIPTION
**Pull request by Benkoff**
resolve #20

### CODE REVIEW

**Comment by danilkuznetsov** [comment]
@Benkoff Please, don't forget about `connect to<number issue>`

File: src/test/java/io/github/externschool/planner/controller/MainControllerTest.java [hunk 1]
```diff
@@ -1,35 +1,41 @@
...
-
+    @Autowired
     private MockMvc mockMvc;
 
-
-    @Before
-    public void setUp() {
-        mockMvc = MockMvcBuilders.standaloneSetup(new MainController()).build();
+    @Test
+    public void testLoginPageReturnsLoginFormView() throws Exception {
```
Comment (danilkuznetsov) [inline src/test/java/io/github/externschool/planner/controller/MainControllerTest.java]: I think in this case, we need to use security configuration.  This page must be available without authorization

File: src/test/java/io/github/externschool/planner/controller/MainControllerTest.java [hunk 2]
```diff
@@ -1,35 +1,41 @@
...
+                .andExpect(view().name("login"))
+                .andExpect(content().contentType("text/html;charset=UTF-8"))
+                .andExpect(content().string(Matchers.containsString("Login Form")));
     }
 
     @Test
-    public void testStartingPageReturnsGreetingView() throws Exception{
-        mockMvc.perform(get("/"))
+    public void testStartingPageReturnsGreetingView() throws Exception {
+        mockMvc.perform(MockMvcRequestBuilders.get("/"))
```
Comment (danilkuznetsov) [inline src/test/java/io/github/externschool/planner/controller/MainControllerTest.java]: For this case, we need to check the auth configuration. This page must be available without authorization

File: src/test/java/io/github/externschool/planner/config/SpringSecurityConfigTest.java [hunk 3]
```diff
@@ -22,7 +22,7 @@
 
     @Test
     public void userIsAuthenticatedTest() throws Exception {
-        mockMvc.perform(formLogin().user("admin@").password("admin"))
+        mockMvc.perform(formLogin().user("admin@a").password("admin"))
```
Comment (danilkuznetsov) [inline src/test/java/io/github/externschool/planner/config/SpringSecurityConfigTest.java]: Also we can check redirection on `/` after successful auth

File: src/main/java/io/github/externschool/planner/config/SpringSecurityConfig.java [hunk 4]
```diff
@@ -6,26 +6,44 @@
...
+                        "/img/**").permitAll()
+                .anyRequest().authenticated()
+                .and()
+
+                .formLogin()
+                .defaultSuccessUrl("/", true)
+                .loginPage("/login")
+                .permitAll()
+                .and()
+                .logout()
```
Comment (danilkuznetsov) [inline src/main/java/io/github/externschool/planner/config/SpringSecurityConfig.java]: it's code from the future, but at the moment it's ok, don't remove it

File: src/test/java/io/github/externschool/planner/config/SpringSecurityConfigTest.java (hunk 3 above)
Comment (Benkoff) [reply src/test/java/io/github/externschool/planner/config/SpringSecurityConfigTest.java]: that redirection to "/" is made by framework without any help of mine. i doubt i'm better than Juergen Hoeller in spring testing

Comment (danilkuznetsov) [reply src/test/java/io/github/externschool/planner/config/SpringSecurityConfigTest.java]: Yes, but if someone changes this way by default, this test will not work

**Pull request merged**